joint-discovery-gatk4-fc.wdl

```diff
@@ -0,0 +1,890 @@
+## Copyright Broad Institute, 2017
+## 
+## This WDL implements the joint discovery and VQSR filtering portion of the GATK 
+## Best Practices (June 2016) for germline SNP and Indel discovery in human 
+## whole-genome sequencing (WGS) and exome sequencing data.
+##
+## Requirements/expectations :
+## - A sample map file with One or more GVCFs produced by HaplotypeCaller in GVCF mode 
+## - Bare minimum 1 WGS sample or 30 Exome samples. Gene panels are not supported.
+##
+## Outputs :
+## - A VCF file and its index, filtered using variant quality score recalibration 
+##   (VQSR) with genotypes for all samples present in the input VCF. All sites that 
+##   are present in the input VCF are retained; filtered sites are annotated as such 
+##   in the FILTER field.
+##
+## Note about VQSR wiring :
+## The SNP and INDEL models are built in parallel, but then the corresponding 
+## recalibrations are applied in series. Because the INDEL model is generally ready 
+## first (because there are fewer indels than SNPs) we set INDEL recalibration to 
+## be applied first to the input VCF, while the SNP model is still being built. By 
+## the time the SNP model is available, the indel-recalibrated file is available to 
+## serve as input to apply the SNP recalibration. If we did it the other way around, 
+## we would have to wait until the SNP recal file was available despite the INDEL 
+## recal file being there already, then apply SNP recalibration, then apply INDEL 
+## recalibration. This would lead to a longer wall clock time for complete workflow 
+## execution. Wiring the INDEL recalibration to be applied first solves the problem.
+##
+## Cromwell version support 
+## - Successfully tested on v29
+## - Does not work on versions < v23 due to output syntax
+##
+## Runtime parameters are optimized for Broad's Google Cloud Platform implementation. 
+## For program versions, see docker containers. 
+##
+## LICENSING : 
+## This script is released under the WDL source code license (BSD-3) (see LICENSE in 
+## https://github.com/broadinstitute/wdl). Note however that the programs it calls may 
+## be subject to different licenses. Users are responsible for checking that they are
+## authorized to run all programs before running this script. Please see the docker 
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
+## licensing information pertaining to the included programs.
+
+workflow JointGenotyping {
+  File unpadded_intervals_file
+
+  String callset_name
+  
+  File ref_fasta
+  File ref_fasta_index
+  File ref_dict
+
+  File dbsnp_vcf
+  File dbsnp_vcf_index
+
+  Array[String] sample_names
+  Array[File] input_gvcfs 
+  Array[File] input_gvcfs_indices 
+
+  String gatk_docker
+
+  Int small_disk
```

Comment:
Should there be some guidelines as to what the values for these disk variables should be?

Reply:
I was thinking users would stick with the default, but would be able change as needed when they hit a disk size block.